src/box/lua/config/configdata.lua

```diff
@@ -775,6 +775,36 @@ local function validate_misplacing(cconfig)
     end
 end
 
+-- Check startup conditions.
+local function validate_startup(instance_name, iconfig_def)
+    local is_startup = type(box.cfg) == 'function'
+    if not is_startup then
+        return
+    end
+
+    local no_snap = snapshot.get_path(iconfig_def) == nil
+    local isolated = instance_config:get(iconfig_def, 'isolated')
+
+    -- Forbid startup without a local snapshot in the isolated
+    -- mode.
+    if no_snap and isolated then
+        error(('Startup failure.\nThe isolated mode is enabled and the ' ..
+            'instance %q has no local snapshot. An attempt to bootstrap ' ..
+            'the database would produce data that has nothing in common ' ..
+            'with data of other replicaset peers: this instance wouldn\'t ' ..
+            'able to join the replicaset later.'):format(instance_name), 0)
```

Comment:
This message seems a bit to long. Maybe something like this will be better:
```
'Startup failure.\nThe isolated mode is enabled and the ' ..
'instance %q has no local snapshot. An attempt to bootstrap ' ..
'the instance would lead to split-brain.'
```

Reply:
Hm. We usually say 'split-brain', when there is some transaction history in common and then it forks. Here we likely have ever different replicaset UUIDs (if it is not set explicitly in the configuration). And conflicting instance IDs. Wouldn't saying just 'split-brain' here be confusing?

I see nothing bad in a long message if it adds useful information and/or directions what to do. I would ever say that I'm tired of error messages that [tells nothing](https://stackoverflow.com/questions/36216437/tarantool-long-wal-write/) to a user.

Reply:
Hoewver, I'm for the shorter message if it fed the same information. If you read my doubts and still prefer to just say 'split-brain', I would do it.

Reply:
Discussed it a bit with @grafin and changed:

```diff
diff --git a/src/box/lua/config/configdata.lua b/src/box/lua/config/configdata.lua
index 587005c23c..7633144f4a 100644
--- a/src/box/lua/config/configdata.lua
+++ b/src/box/lua/config/configdata.lua
@@ -790,9 +790,8 @@ local function validate_startup(instance_name, iconfig_def)
     if no_snap and isolated then
         error(('Startup failure.\nThe isolated mode is enabled and the ' ..
             'instance %q has no local snapshot. An attempt to bootstrap ' ..
-            'the database would produce data that has nothing in common ' ..
-            'with data of other replicaset peers: this instance wouldn\'t ' ..
-            'able to join the replicaset later.'):format(instance_name), 0)
+            'the instance would lead to the split-brain situation.'):format(
+            instance_name), 0)
     end
 
     -- TODO: There is a situation, which looks similar, but we
diff --git a/test/config-luatest/isolated_mode_test.lua b/test/config-luatest/isolated_mode_test.lua
index 46f4a88bc7..d3a7cf6283 100644
--- a/test/config-luatest/isolated_mode_test.lua
+++ b/test/config-luatest/isolated_mode_test.lua
@@ -44,9 +44,8 @@ g.test_startup_no_snap = function()
     -- Verify the exit code and the error reported to stderr.
     local exp_err = 'Startup failure.\n' ..
         'The isolated mode is enabled and the instance "i-003" has no local ' ..
-        'snapshot. An attempt to bootstrap the database would produce data ' ..
-        'that has nothing in common with data of other replicaset peers: ' ..
-        'this instance wouldn\'t able to join the replicaset later.'
+        'snapshot. An attempt to bootstrap the instance would lead to the ' ..
+        'split-brain situation.'
     t.assert_covers(res, {
         exit_code = 1,
         stderr = ('LuajitError: %s\nfatal error, exiting the event loop')
```

Also, rebased the patch on top of the current master.